src/dom/DomUtil.js

```diff
@@ -100,45 +100,38 @@ export function getPosition(el) {
 	return el._leaflet_pos || new Point(0, 0);
 }
 
+const documentStyle = document.documentElement.style;
+// Safari still needs a vendor prefix, we need to detect with property name is supported.
+const userSelectProp = ['userSelect', 'WebkitUserSelect'].find(prop => prop in documentStyle);
+let prevUserSelect;
+
 // @function disableTextSelection()
-// Prevents the user from generating `selectstart` DOM events, usually generated
-// when the user drags the mouse through a page with text. Used internally
+// Prevents the user from selecting text in the document. Used internally
 // by Leaflet to override the behaviour of any click-and-drag interaction on
 // the map. Affects drag interactions on the whole document.
+export function disableTextSelection() {
+	const value = documentStyle[userSelectProp];
+	if (value !== 'none') {
+		prevUserSelect = value;
+		documentStyle[userSelectProp] = 'none';
+	}
+}
 
 // @function enableTextSelection()
 // Cancels the effects of a previous [`L.DomUtil.disableTextSelection`](#domutil-disabletextselection).
-export let disableTextSelection;
-export let enableTextSelection;
-let _userSelect;
-if ('onselectstart' in document) {
-	disableTextSelection = function () {
-		DomEvent.on(window, 'selectstart', DomEvent.preventDefault);
-	};
-	enableTextSelection = function () {
-		DomEvent.off(window, 'selectstart', DomEvent.preventDefault);
-	};
-} else {
-	disableTextSelection = function () {
-		const style = document.documentElement.style;
-		_userSelect = style.userSelect;
-		style.userSelect = 'none';
-	};
-	enableTextSelection = function () {
-		document.documentElement.style.userSelect = _userSelect;
-		_userSelect = undefined;
-	};
+export function enableTextSelection() {
+	documentStyle[userSelectProp] = prevUserSelect;

```

Comment:
If `enableTextSelection` is called before `disableTextSelection` it will overwrite the default value `auto`. We should set `prevUserSelect` to `auto` or add a check before setting the value

Reply:
Ok it doesn't allow to set invalid values but I still think it would be cleaner

Reply:
Agreed, I'll add a check :+1: